daemon/containerd/image_push.go

```diff
@@ -163,14 +166,35 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 	}
 
 	err = remotes.PushContent(ctx, pusher, target, store, limiter, platforms.All, handlerWrapper)
-	if err != nil {
-		if containerdimages.IsIndexType(target.MediaType) && cerrdefs.IsNotFound(err) {
+	if err != nil && containerdimages.IsIndexType(target.MediaType) {
+		if cerrdefs.IsNotFound(err) && platform == nil {
+			// If push failed because of a missing content, no specific platform was requested
+			// and the target is an index, select a platform-specific manifest to push instead.
+			target, err = i.getPushDescriptor(ctx, img, nil)
```

Comment:
Should we check the target is different here? It seems to be relying the availability check preventing that but not sure we can guarantee that.

Reply:
Makes sense to me. Done, thanks!